motion/joystick_guidance/scripts/joystick_guidance.py

```diff
@@ -38,6 +52,95 @@ def callback(self, msg):
 
 		self.pub.publish(joystick_msg)
 
+	def calculate_joystick_point(joystick_msg):
+		"""
+		Calculates a point in the local frame based on the given thrust-vectors 
+		from joystick 
+
+		If required it must be converted into the global reference-frame
+		"""
+
+		# Scaling the force to get a linearized model
+		scaled_force_x, scaled_force_y, scaled_force_z = scale_force_vectors(joystick_msg)
+
+		# TODO Change this to the required point-type size
+		calculated_point = [scaled_force_x, scaled_force_y, scaled_force_z]
+
+		# Calculating the length of the vectors
+		vector_length_square = 0
+		for i in range(len(calculated_point)):
+			vector_length_square += pow(calculated_point(i), 2)
+
+		# Scale the point to hold withing the designated area
+		if vector_length_square < pow(self.min_point_range, 2):
+			# Under the set limit. Setting current vector to 0
+			vector_length_square *= 0
+		elif vector_length_square >= pow(self.max_point_range, 2):
+			# Over the set limit. Normalizing
+			calculated_point /= sqrt(vector_length_square)
+
+		return calculated_point
+
+
+	def scale_force_vectors(joystick_msg):
+        """
+        Scales the thrust-vectors to
+        """
+
+        # Recovering the given forces
+		scale_x_vec = joystick_msg.force.x
+		scale_y_vec = joystick_msg.force.y
+		scale_z_vec = joystick_msg.force.z
+
+        # Using a try-catch to prevent out-of-bounds to become a large problem
+        try:
+            # Scale each force
+            for i in range(len(self.limits)):
+                if i == 0 and scale_x_vec < self.limits[i]:
```

Comment:
Everything inside the for-loop should be in a helper function, since the code structure is identical for all three loops